src/smt/smt_engine.cpp

```diff
@@ -2185,6 +2185,25 @@ void SmtEngine::setDefaults() {
         << endl;
     options::bvLazyRewriteExtf.set(false);
   }
+
+
+  if (options::skeletonPreprocessing()
+      &&  options::arithMLTrick())
+  {
+    if (options::skeletonPreprocessing.wasSetByUser()
+        && options::arithMLTrick.wasSetByUser())
+    {
+      throw OptionException(std::string(
+          "Cannot use MipLibTrick when using cryptominisat instead of circuit"
+          "propagators. Try turn off --skeleton-preprocessing"));
+    }
+    else if (options::skeletonPreprocessing.wasSetByUser()){
+          setOption("skeleton-preprocessing", false);
+    }
+    else if (options::arithMLTrick.wasSetByUser()){
+          setOption("miplib-trick", false);
+    }
+  }
```

Comment:
Also add a check that complains when `options::skeletonPreprocessing()` is set but [`isBuiltWithCryptominisat()`](https://github.com/CVC4/CVC4/blob/master/src/base/configuration.h#L96) returns false.